### ISSUE TITLE
Date / time format in reports (beginTime, endTime) is neither user-friendly nor parser-friendly

### ISSUE THREAD

**Issue by crazy4chrissi** [duplicati/duplicati#2753]
In reports, Duplicati includes the BeginTime and EndTime. Building a parser that parses these reports (see issue #2411), I noticed that Duplicati does not use a consistent format, but seems to use the format defined at the operating system (?). This makes it really hard to parse these dates.
Some examples from what I saw:

> dd.mm.YYYY HH:ii:ss
> m/dd/YYYY HH:ii:ss AM/PM
> dd/mm/YYYY HH:ii:ss
> dd-mm-yy HH:ii:ss
> dd-m-YYYY HH:ii:ss

Moreover, the times in the reports are not user-friendly, as they are always in UTC instead of the user's local timezone.

I'd propose to:
- print beginTime and endTime in the user's timezone
- add a parser-friendly time, e.g. as a unix timestamp

> beginTime: 2017/09/22 16:23:00 CEST
> beginTimestamp: 1506090180

or as one line:

> beginTime: 2017/09/22 16:23:00 CEST (1506090180)

**Comment by dgcom** [comment]
I suggest for all date **logging** purposes use ISO 8601 recommendations (https://en.wikipedia.org/wiki/ISO_8601)
And UI should present date/time in locale-specific formatting.

This way all logs and reports can be machine-parsable without sacrificing user experience.

**Comment by crazy4chrissi** [comment]
I'd be fine with this. Even though I don't like the T delimiter in ISO 8601 much.

**Comment by daylightbrightledlight** [comment]
I would prefer using the following:
> `2017-09-22T17:34:00-04:00`

The `2017-09-22` represents `YYYY-MM-DD`
The `T` separates the date from the time.
The `17:34:00` represents 5:34:00 PM EDT which is `-04:00` away from UTC.

If it was using UTC, it would be the following:
> `2017-09-22T21:34:00+00:00`

This kind of format would make it incredibly easy and consistent to parse, as it would have a fixed length.

**Comment by crazy4chrissi** [comment]
@daylightbrightledlight Isn't this ISO 8601?

**Comment by davegold24** [comment]
I may be hijacking this, in which case someone please tell me to open a new issues.
I noticed in the daily email that LastBackupDate is in local time, but EndTime and StartTime are in UTC.

Could it be a bad thing to switch from UTC to local for the report? 
I'm specifically thinking updating Duplicati/Library/Main/ResultClasses.cs from 
EndTime = DateTime.UtcNow 
to
EndTime = DateTime.Now

and
this.BeginTime = DateTime.UtcNow
to
this.BeginTime = DateTime.Now

I'm not a C# programmer, but per https://msdn.microsoft.com/en-us/library/system.datetime.utcnow(v=vs.110).aspx, it looks like this would modify the daily report.

Any thoughts/objections to starting with that, and separating the logging format into a separate issue?

My only concern is if by modifying what I propose that there are some side effects I'm not aware of.
-Dave

**Comment by crazy4chrissi** [comment]
No :( Please don't just change everything to local time zone. Or if you do, at least add what time zone this is.

I see several issues: 
- A company can have hundreds of systems all over the world in a dozen different time zones. Imagine you are the guy who gets all the reports and needs to convert all these time zones in your head. This is a nightmare, especially if the report does not mention which time zone is the local time zone of this system.
(This scenario is not artificial, at www.duplicati-monitoring.com we see users like that.)
- External tools that parse the reports like www.duplicati-monitoring.com and probably also dupReport currently assume these times are in UTC. If you change this, you need to somehow indicate that it is not UTC anymore. Adding the timezone offset would be a good idea here.
- Mail reports are currently hard to read for humans anyway. For example, sizes given as number of bytes is not user friendly. So I don't think it makes sense to make one thing user friendly but impossible to parse while keeping everything else parser-friendly but hard to read for humans.

**Comment by davegold24** [comment]
Hi @crazy4chrissi Good points. I'd like to suggest two changes then:
1. LastBackupTime be reported in UTC
2. LastBackupTime, BeginTime and EndTime all be labeled UTC. Having them not be labeled is as confusing as having them report in non-local time.

In response to your specific issues:
1. I'm impressed you are using Duplicati at that scale. How do you centrally manage it?
2. Do you have more info on duplicati-monitoring? I see the web site, but I don't see any example reports.
3. Agreed on mail reports being difficult to read. I've been working on one that is more detailed, and it I'm going to take your feedback into account. dupReport already has that feature, but a more friendly report is certainly important.

Just to reinforce--I'm not not looking for easy to read per se, just consistent.

With that in mind, any disagreement on:
1. Adding "UTC" to the BeginTime and EndTime lines?
2. Changing LastBackupTime to UTC and labeling it as such?

I think a more thorough option would be useful--have an option to report in the default UTC time, or localtime as an option.

Anyways, thoughts are appreciated. Certainly don't want to break your service!

--Dave

**Comment by crazy4chrissi** [comment]
I agree that consistency is important and having all times in UTC and labeled as such would be fine for me. The main concern I had when opening this issue was more the date format, not the time zone. Currently, the format depends on the system. This makes it really hard to parse. It should instead be a standard format, independent of the system settings.
As proposed before, I would prefer dates in some standardized format like ISO 8601. Then, I would not care about the time zone, as ISO 8601 includes the time zone offset.

1. We are not using Duplicati at this scale ourselves. We offer the www.duplicati-monitoring.com service, and we see users that monitor hundreds of backups using our service. So our monitoring service seems to be one part of managing Duplicati at this scale.
2. Regarding our monitoring service: If you have not seen it yet, issue #2411 might be interesting for you or this [forum thread](https://forum.duplicati.com/t/easier-to-read-backup-reports-or-weekly-summarization/718/6). We currently have almost no real documentation of our service ;) Just try it out. Our email reports look neat and are user friendly, I promise.

**Comment by mmiat** [comment]
it could be useful an option "show time with your local time" or similar
i run backup at midnight, and report show me that i run backup the day before...
and when daylight change, the discomfort annoyance is even greater

**Comment by crazy4chrissi** [comment]
The changelog says:

> 2017-11-22 - 2.0.2.13_canary_2017-11-22
> * Timestamps in the log are now reported in local time and include a UTC timestamp for easy script parsing

I saw a user on duplicati-monitoring whose canary version reported "28-11-2017 12:59:06 (1511870346)". This is okay for me.

So in my opinion, this issue can be closed. Other opinions?

**Comment by crazy4chrissi** [comment]
Anyone who has problems with the new formatting of date/time should open a new issue. I think now it is parser-friendly and more user-friendly, so I consider this issue fixed.
